lib/autokey/configmanager/configmanager.py

```diff
@@ -669,22 +669,42 @@ def check_hotkey_unique(self, modifiers, hotKey, newFilterPattern, targetItem):
         @param newFilterPattern:
         @param targetItem: the phrase for which the hotKey to be used
         """
-        for item in self.allFolders:
-            if model.TriggerMode.HOTKEY in item.modes:
-                if item.modifiers == modifiers and item.hotKey == hotKey and item.filter_matches(newFilterPattern):
-                    return item is targetItem, item
+        item = self.get_item_with_hotkey(modifiers, hotKey, newFilterPattern)
+        if item:
+            return item is targetItem, item
+        else:
+            return True, None
 
-        for item in self.allItems:
-            if model.TriggerMode.HOTKEY in item.modes:
-                if item.modifiers == modifiers and item.hotKey == hotKey and item.filter_matches(newFilterPattern):
-                    return item is targetItem, item
+    def get_item_with_hotkey(self, modifiers, hotKey, newFilterPattern=None):
+        """
+        Gets first item with the specified hotkey. Also checks the
+        special hotkeys configured from the advanced settings dialog.
+        Checks folders first, then phrases, then special hotkeys.
+
+        @param modifiers: modifiers for the hotkey
+        @param hotKey: the hotkey to check
+        @param newFilterPattern:
+        """
+        for searchSpace in [self.allFolders, self.allItems]:
```

Comment:
You _could_ use `for item in itertools.chain(self.allFolders, self.allItems)` to remove one line of nesting here, if you wish. The chain will simply iterate over the items in both lists.

Reply:
Neat python, thanks!

Reply:
Resolved. Just got one failing test to fix, then I'll push.